backend/src/apiserver/client_manager/client_manager.go

```diff
@@ -166,54 +182,76 @@ func (c *ClientManager) Authenticators() []auth.Authenticator {
 	return c.authenticators
 }
 
-func (c *ClientManager) init() {
-	glog.Info("Initializing client manager")
-	glog.Info("Initializing DB client...")
-	db := InitDBClient(common.GetDurationConfig(initConnectionTimeout))
-	db.SetConnMaxLifetime(common.GetDurationConfig(dbConMaxLifeTime))
-	glog.Info("DB client initialized successfully")
-	// time
-	c.time = util.NewRealTime()
-
-	// UUID generator
-	c.uuid = util.NewUUIDGenerator()
+func (c *ClientManager) ControllerClient() ctrlclient.Client {
+	return c.controllerClient
+}
 
-	c.db = db
-	c.experimentStore = storage.NewExperimentStore(db, c.time, c.uuid)
-	c.pipelineStore = storage.NewPipelineStore(db, c.time, c.uuid)
-	c.jobStore = storage.NewJobStore(db, c.time)
-	c.taskStore = storage.NewTaskStore(db, c.time, c.uuid)
-	c.resourceReferenceStore = storage.NewResourceReferenceStore(db)
-	c.dBStatusStore = storage.NewDBStatusStore(db)
-	c.defaultExperimentStore = storage.NewDefaultExperimentStore(db)
-	glog.Info("Initializing Object store client...")
-	c.objectStore = initMinioClient(common.GetDurationConfig(initConnectionTimeout))
-	glog.Info("Object store client initialized successfully")
-	// Use default value of client QPS (5) & burst (10) defined in
-	// k8s.io/client-go/rest/config.go#RESTClientFor
-	clientParams := util.ClientParameters{
-		QPS:   common.GetFloat64ConfigWithDefault(clientQPS, 5),
-		Burst: common.GetIntConfigWithDefault(clientBurst, 10),
+func (c *ClientManager) init(onlyKubernetesWebhookMode bool) {
+	glog.Info("Initializing controller client...")
+	restConfig, err := util.GetKubernetesConfig()
+	if err != nil {
+		panic(fmt.Sprintf("Failed to initialize the RestConfig: %v", err))
 	}
+	controllerClient, err := ctrlclient.New(
+		restConfig, ctrlclient.Options{Scheme: scheme},
+	)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to initialize the controller client: %v", err))
+	}
+	c.controllerClient = controllerClient
+	glog.Info("Controller client initialized successfully.")
+
+	if !onlyKubernetesWebhookMode {
```

Comment:
Rather than indent, it'd be easier to read with something like:

```python
if onlyKubernetesWebhookMode {
    return
}

glog.Info("Initializing client manager")
....
```